skrf/circuit.py

```diff
@@ -785,9 +785,47 @@ def s_external(self) -> np.ndarray:
             Scattering parameters of the circuit for the external ports.
             Shape `f x nb_ports x nb_ports`
         """
+        # Global matrix version by indexing into s matrix
+        # port_indexes = self.port_indexes
+        # a, b = np.meshgrid(port_indexes, port_indexes, indexing='ij')
+        # S_ext = self.s[:, a, b]
+
+        # generate index lists of internal and external ports
```

Comment:
```suggestion
        # Instead of calculating all S-parameters and taking a submatrix,
        # the following faster approach only calculates external the S-parameters
        # from block-matrix operations.
        
        # generate index lists of internal and external ports
```